### PR TITLE
feat(BivariateColorManager): 10841 - added Legend to BivariateColorMa…

### PR DESCRIPTION
**Pull request by propakov**
- PopupTooltip moved to CommonRoutesFeatures
- BivariateLegend wrapper extended with new props to customize ui-kit legend
- bivariateColorManagerAtom with layers selection handling
- Sublist moved to separate component
Screenshot 2022-07-28 at 10 36 56

### CODE REVIEW

File: src/features/bivariate_color_manager/components/LegendWithMap/LegendWithMap.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        direction: directions.horizontal,
+      },
+      x: {
+        label: verticalLabel,
+        direction: directions.vertical,
+      },
+    },
+  };
+
+  const renderXAxisLabel = (_bivAxis, rootClassName) => (
```
Comment (IlyaIzr) [inline src/features/bivariate_color_manager/components/LegendWithMap/LegendWithMap.tsx]: would be cool to hang on types for arguments here

File: src/features/bivariate_color_manager/components/LegendWithMap/LegendWithMap.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      <div className={s.axisDirectionLabel}>
+        {directions.horizontal.map((direction, i) => (
+          <div key={i}>{joinAndCapitalizeItems(direction)}</div>
+        ))}
+      </div>
+      {horizontalLabel}
+    </div>
+  );
+
+  const renderYAxisLabel = (_bivAxis, rootClassName) => (
```
Comment (IlyaIzr) [inline src/features/bivariate_color_manager/components/LegendWithMap/LegendWithMap.tsx]: would be cool to hang on types for arguments here too

File: src/features/bivariate_color_manager/components/Sublist/Sublist.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      appear
+      classNames={fadeClassNames}
+    >
+      {(ref) => (
+        <tr ref={ref} className={s.sublist}>
+          <td />
+          <td />
+          <td>{verticalList.map(renderVerticalItem)}</td>
+          <td>{horizontalList.map(renderHorizontalItem)}</td>
+        </tr>
```
Comment (IlyaIzr) [inline src/features/bivariate_color_manager/components/Sublist/Sublist.tsx]: why do we need table tags, especially empty ones?

Comment (propakov) [reply src/features/bivariate_color_manager/components/Sublist/Sublist.tsx]: we have 4 columns, and we need to locate sublists on last 2 columns, first 2 should remain empty.
Screenshot 2022-07-28 at 15 06 15

maybe there is more clear way of how to skip first 2 columns idk
anyway feel free to share of how to make it clearer :)

File: src/features/bivariate_color_manager/components/LegendWithMap/LegendWithMap.tsx (hunk 1 above)
Comment (propakov) [reply src/features/bivariate_color_manager/components/LegendWithMap/LegendWithMap.tsx]: will do that! 👍

File: src/features/bivariate_color_manager/components/Sublist/Sublist.tsx (hunk 3 above)
Comment (IlyaIzr) [reply src/features/bivariate_color_manager/components/Sublist/Sublist.tsx]: I see, probaply it's the easiest way. However I'd call Sublist component with a less abstract name as it's not a shared component - like what it does\shows us? Availible layers? Featured layers? Expandable list?

**Comment by IlyaIzr** [comment]
I'd also add `cursor: pointer` to all expandable rows

File: src/features/bivariate_color_manager/components/Sublist/Sublist.tsx (hunk 3 above)
Comment (propakov) [reply src/features/bivariate_color_manager/components/Sublist/Sublist.tsx]: if to think from entities side it's a list of sentiments combinations
so maybe SentimentsCombinationsList?

Comment (propakov) [reply src/features/bivariate_color_manager/components/Sublist/Sublist.tsx]: oh, god, our root table component has this name  🤔

**Review by IlyaIzr — APPROVED** [review]

**Pull request merged**